### PR TITLE
resolve Wconversion related warnings

### PR DESCRIPTION
**Pull request by timaydin**
Description
-----------
Avoid warnings produced as a result of -Wconversion option to gcc

### CODE REVIEW

**Comment by moninom1** [comment]
Hi @timaydin 

There are some CI checks failure, can you please help in checking them.

Thank You

File: source/FreeRTOS_TCP_Reception.c [hunk 1]
```diff
@@ -331,14 +331,14 @@
                          */
                         if( pucPtr[ 0U ] == tcpTCP_OPT_SACK_A )
                         {
-                            ucLen -= 2U;
+                            ucLen -= 2;
```
Comment (AniruddhaKanhere) [inline source/FreeRTOS_TCP_Reception.c:334]: I am unsure about this change. `ucLen` is unsigned. Then while assigning 2 to it, why not make 2 explicitly signed by using `U`?

Comment (moninom1) [reply source/FreeRTOS_TCP_Reception.c:334]: This will also give MISRA violation.

File: source/FreeRTOS_TCP_Reception.c [hunk 2]
```diff
@@ -331,14 +331,14 @@
...
                             lIndex += 2;
 
                             while( ucLen >= ( uint8_t ) 8U )
                             {
                                 prvReadSackOption( pucPtr, ( size_t ) lIndex, pxSocket );
                                 lIndex += 8;
-                                ucLen -= 8U;
+                                ucLen -= 8;
```
Comment (moninom1) [inline source/FreeRTOS_TCP_Reception.c:341]: This is not solving the issue . Earlier warning was "**conversion from ‘unsigned int’ to ‘uint8_t**’" with your fix the warning has now changed to **conversion from ‘int’ to ‘uint8_t’**

**Comment by moninom1** [comment]
Hi @timaydin  I was reproducing the issue and i can see that there are 40+ warning with enabling Wconverstion. Were you seeing the same number of issue?  If not can you let us know how you reproduced the issue?

**Pull request closed without merging**